openBCIBoard.js

```diff
@@ -856,180 +1013,174 @@ function OpenBCIFactory() {
+                        this.sync.ntp2 = now();
+                        this._writeAndDrain('<' + (this.sync.ntp1 * 1000) + (this.sync.ntp2 * 1000));
+                        this.searchingBuf = this.searchBuffers.miscStop;
+
                     } else {
                         getChannelSettingsObj(data.slice(i)).then((channelSettingsObject) => {
                             this.emit('query',channelSettingsObject);
                         }, (err) => {
                             console.log('Error: ' + err);
                         });
-                        this.searchingBuf = this.moneyBuf;
+                        this.searchingBuf = this.searchBuffers.miscStop;
                         break;
                     }
                 }
             }
         } else { // steaming operation should lead here...
-            // send input data to master buffer
-            this._bufMerger(data);
-
-            // parse the master buffer
-            while(this.masterBuffer.packetsRead < this.masterBuffer.packetsIn) {
-                var rawPacket = this._bufPacketStripper();
-                var newSample = openBCISample.convertPacketToSample(rawPacket);
-                if(newSample) {
-                    this.emit('rawDataPacket', rawPacket);
-                    newSample._count = this.sampleCount++;
-                    if(this.impedanceTest.active) {
-                        if (this.impedanceTest.onChannel != 0) {
-                            // Get an average of the impedance
-                            openBCISample.impedanceCalculationForChannel(newSample,this.impedanceTest.onChannel)
-                                .then(rawValue => {
-                                    impedanceTestApplyRaw.call(this,rawValue);
-                                }).catch(err => {
-                                    console.log('Impedance calculation error: ' + err);
+
+            // cbuffer
+            var bytesToRead = sizeOfData;
+
+            // is there old data?
+            if (this.buffer) {
+                // Get size of old buffer
+                var oldBufferSize = this.buffer.byteLength;
+
+                // Make a new buffer
+                var newDataBuffer = new Buffer(bytesToRead + oldBufferSize);
+
+                // Put old buffer in the front of the new buffer
+                var oldBytesWritten = this.buffer.copy(newDataBuffer);
+
+                // Move the incoming data into the end of the new buffer
+                data.copy(newDataBuffer,oldBytesWritten);
+
+                // Over write data
+                data = newDataBuffer;
+
+                // Update the number of bytes to read
+                bytesToRead += oldBytesWritten;
+            }
+
+            var readingPosition = 0;
+
+            // 45 < (200 - 33) --> 45 < 167 (good) | 189 < 167 (bad) | 0 < (28 - 33) --> 0 < -5 (bad)
```

Comment:
comment about why you need to deal with this. because of serialport, the data packet can be split